# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/credentialmanagement/identity/WebIdentityChild.cpp
# Commit: 46af55564501
# Full Hash: 46af555645013a126416a440b658f2969eca4b20
# Author: Benjamin VanderSloot <bvandersloot@mozilla.com>
# Date: 2025-06-06 08:52:03
# Regressor Bug: 1969017
# File Overlap Count: 2
# Description:
#   Bug 1969017 - Refactor FedCM to use a managed actor rather than tagging onto WindowGlobalParent - r=anti-tracking-reviewers,emz
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D251617
# ==============================================================================

diff -r 55c4afa3a46c -r 46af55564501 dom/credentialmanagement/identity/WebIdentityChild.cpp
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/dom/credentialmanagement/identity/WebIdentityChild.cpp	Thu Jun 05 14:53:36 2025 +0000
@@ -0,0 +1,23 @@
+/* -*- Mode: C++; tab-width: 8; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
+/* vim: set ts=8 sts=2 et sw=2 tw=80: */
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "mozilla/dom/WebIdentityChild.h"
+#include "mozilla/dom/WebIdentityHandler.h"
+
+namespace mozilla::dom {
+
+void WebIdentityChild::ActorDestroy(ActorDestroyReason why) {
+  if (mHandler) {
+    mHandler->ActorDestroyed();
+    mHandler = nullptr;
+  }
+}
+
+void WebIdentityChild::SetHandler(WebIdentityHandler* aHandler) {
+  mHandler = aHandler;
+}
+
+}  // namespace mozilla::dom